phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
All of the language methods that you mention take a SINGLE entry and process it.
What we would need is a method that would take either a string, an asoc-array (key->array of params) or an array of keys or array of key+params...  This for me is kind of complex parsing, that of course would use the above services, but needs some significant code.

I don't see that much of an issue with extension messages: the requirement is that the appropriate language file is loaded within the "unnamed new method".  And given that that must be done there (the core cannot have all the possible reasons for an extension not being enableable), for the extension itself would not be complex to process the language strings, using the methods you mention, and without the need of more complex processing in the core.

I understand that we can do what we need with values and types, but I honestly believe that things are much easier to understand and comprehend by the general user when "make sense".  So, having "true" being OK, and having 'some string' being KO does not make much sense to me...

About exceptions, although I think this would be a good solution, it has a significant limitation: single message.